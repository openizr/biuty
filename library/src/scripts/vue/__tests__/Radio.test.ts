/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UIRadio from 'scripts/vue/Radio.vue';

type component = any; // eslint-disable-line @typescript-eslint/no-explicit-any

jest.mock('scripts/helpers/markdown');
jest.mock('scripts/helpers/generateRandomId');
const options: { value: string; label: string; disabled?: boolean; }[] = [
  { value: 'option1', label: 'Option 1' },
  { value: 'option2', label: 'Option 2' },
  { value: 'option3', label: 'Option 3' },
  { value: 'option4', label: 'Option 4' },
];

describe('vue/UIRadio', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', (done) => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', options, modifiers: 'large' },
    });
    // Ensures value prop's reactivity.
    wrapper.setProps({ value: 'option1' });
    wrapper.vm.$nextTick().then(() => {
      wrapper.find('input').trigger('focus');
      wrapper.find('input').trigger('change');
      expect(wrapper.html()).toMatchSnapshot();
      done();
    });
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', options, id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with label', () => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', options, label: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with helper', () => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', options, helper: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with value', () => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', value: 'option1', options },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    const wrapper = mount(UIRadio, {
      propsData: { name: 'test', modifiers: 'disabled', options },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with option disabled', () => {
    const onChange = jest.fn();
    const wrapper = mount(UIRadio, {
      propsData: {
        name: 'test', onChange, modifiers: 'disabled', options: [{ value: 'option5', label: 'Option 5', disabled: true }],
      },
    });
    wrapper.find('input').trigger('change');
    expect(wrapper.html()).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledTimes(0);
  });

  test('renders correctly - with listeners', async () => {
    const onChange = jest.fn();
    const onFocus = jest.fn();
    const wrapper = mount(UIRadio, {
      propsData: {
        name: 'test', options, value: 'option2',
      },
      listeners: {
        focus: onFocus,
        change: onChange,
      },
    });
    await (wrapper.vm as component).focusField('option2');
    wrapper.find('input').setChecked(true);
    expect(wrapper.html()).toMatchSnapshot();
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith('option2');
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('option1');
  });
});
