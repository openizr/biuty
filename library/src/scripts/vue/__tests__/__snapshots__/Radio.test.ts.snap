// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UIRadio renders correctly - basic 1`] = `
"<div class=\\"ui-radio ui-radio--large\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option ui-radio__wrapper__option--checked\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - disabled 1`] = `
"<div class=\\"ui-radio ui-radio--disabled\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"-1\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"-1\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"-1\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"-1\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - with helper 1`] = `
"<div class=\\"ui-radio\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div> <span classname=\\"ui-radio__wrapper__helper\\">test</span>
</div>"
`;

exports[`vue/UIRadio renders correctly - with id 1`] = `
"<div id=\\"test\\" class=\\"ui-radio\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - with label 1`] = `
"<div class=\\"ui-radio\\"><label for=\\"z8a8d7a5zad4\\" class=\\"ui-radio__label\\">test</label>
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - with listeners 1`] = `
"<div class=\\"ui-radio\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option ui-radio__wrapper__option--checked\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - with option disabled 1`] = `
"<div class=\\"ui-radio ui-radio--disabled\\" onchange=\\"function () {
        return fn.apply(this, arguments);
      }\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option ui-radio__wrapper__option--disabled\\"><input name=\\"test\\" type=\\"radio\\" disabled=\\"disabled\\" tabindex=\\"-1\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option5\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 5</span></label></div>
  <!---->
</div>"
`;

exports[`vue/UIRadio renders correctly - with value 1`] = `
"<div class=\\"ui-radio\\">
  <!---->
  <div class=\\"ui-radio__wrapper\\"><label class=\\"ui-radio__wrapper__option ui-radio__wrapper__option--checked\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option1\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 1</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option2\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 2</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option3\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 3</span></label><label class=\\"ui-radio__wrapper__option\\"><input name=\\"test\\" type=\\"radio\\" tabindex=\\"0\\" class=\\"ui-radio__wrapper__option__radio\\" value=\\"option4\\"> <span classname=\\"ui-radio__wrapper__option__label\\">Option 4</span></label></div>
  <!---->
</div>"
`;
